src/twisted/words/protocols/irc.py

```diff
@@ -2651,8 +2654,17 @@ def dataReceived(self, data):
 
     def lineReceived(self, line):
         if bytes != str and isinstance(line, bytes):
-            # decode bytes from transport to unicode
-            line = line.decode("utf-8")
+            # decode bytes from transport to str
+            for codec in self.decode_codecs:
+                try:
+                    line = line.decode(codec)
+                    break
+                except ValueError:
+                    log.msg(f"Couldn't decode line '{line}' with codec {codec}")
```

Comment:
We definitely want to use `repr()` on the line here, as it could contain control characters etc.

```suggestion
                    log.msg(f"Couldn't decode line {line!r} with codec {codec}")
```

I'd also say this should be a debug message but the logging in this module hasn't been updated to use `twisted.logger` yet so no need to make that change right now.